### PR TITLE
#694 don't show blocked users or users that blocked you in group chat

### PR DESCRIPTION
**Pull request by JorisBassle**

### CODE REVIEW

File: apps/backend/src/features/api/api.service.ts [hunk 1]
```diff
@@ -445,6 +445,18 @@ export class ApiService {
...
 
+    /**
+     * Check if you are blocked by the other twin.
+     */
+    async checkIfBlocked({ location, userId }: { location: string; userId: string }): Promise<boolean> {
+        const url = `http://[${location}]/api/v2/blocked/${userId}`;
+        try {
+            return (await axios.get(url)).data;
+        } catch {
+            return;
```
Comment (EdwardKerckhof) [inline apps/backend/src/features/api/api.service.ts:456]: return false if request did not go through

File: apps/backend/src/features/chat/chat.service.ts [hunk 2]
```diff
@@ -60,6 +62,9 @@ export class ChatService {
             const msg = await this._messageService.createMessage(createGroupChatDTO.messages[0]);
             Promise.all(
                 chat.parseContacts().map(async c => {
+                    const isBlocked = await this._blockedContactService.isBlocked({ userId: c.id });
+                    const iAmBlocked = await this._apiService.checkIfBlocked({ userId, location: c.location });
```
Comment (EdwardKerckhof) [inline apps/backend/src/features/chat/chat.service.ts:66]: change iAmBlocked to isCurrentUserBlocked

File: apps/backend/src/features/message/states/system-message.state.ts [hunk 3]
```diff
@@ -19,12 +20,17 @@ export class AddUserSystemState implements SubSystemMessageState {
...
         private readonly _chatGateway: ChatGateway,
-        private readonly _messageService: MessageService
+        private readonly _messageService: MessageService,
+        private readonly _blockedContactService: BlockedContactService
     ) {}
 
     async handle({ message, chat }: { message: MessageDTO<SystemMessage>; chat: ChatDTO }): Promise<unknown> {
         const { contact, adminLocation } = message.body as GroupUpdate;
         const userId = this._configService.get<string>('userId');
+        const iAmBlocked = await this._apiService.checkIfBlocked({ userId, location: contact.location });
```
Comment (EdwardKerckhof) [inline apps/backend/src/features/message/states/system-message.state.ts:30]: change iAmBlocked to isCurrentUserBlocked

File: apps/frontend/src/components/GroupManagement.vue [hunk 4]
```diff
@@ -3,10 +3,10 @@
         <!--<h3 class="mt-2 ml-2 text-base text-left mb-4">Actions</h3>-->
         <div class="flex xl:grid xl:grid-cols-2 w-full gap-y-2 items-center justify-around xl:justify-items-center">
             <!--<div v-if="!chat.isGroup" class="flex flex-col px-2 py-4 w-full cursor-pointer">
-                <i class="fas fa-minus-circle m-3"></i>
-                <p class="text-xs text-gray-300">Bio</p>
-                <p class="text-xs text-gray-500">Block user</p>
-            </div>-->
+<i class="fas fa-minus-circle m-3"></i>
```
Comment (EdwardKerckhof) [inline apps/frontend/src/components/GroupManagement.vue:6]: prettier?

File: apps/frontend/src/components/GroupManagement.vue [hunk 5]
```diff
@@ -16,13 +16,13 @@
...
             </div>
             <!--<div 
-                v-if="!blocked"
-                class="flex flex-col bg-white text-primary items-center px-2 justify-center w-24 h-16 rounded-lg border border-gray-300 cursor-pointer"
-                @click="$emit('app-mute')"
-            >
-                <span class="material-symbols-rounded"> notifications_off </span>
-                <p class="text-xs mt-1">{{ chat.isMute ? 'Unmute' : 'Mute' }}</p>
-            </div>-->
+v-if="!blocked"
```
Comment (EdwardKerckhof) [inline apps/frontend/src/components/GroupManagement.vue:19]: prettier?

File: apps/frontend/src/store/blockStore.ts [hunk 6]
```diff
@@ -22,3 +24,16 @@ export const removeUserFromBlockList = (userId: string) => {
 };
 
 export const userIsBlocked = (userId: string) => blocklist.value.includes(userId);
+
+export const iAmBlocked = async (location: string, userId: string): Promise<boolean> => {
```
Comment (EdwardKerckhof) [inline apps/frontend/src/store/blockStore.ts:28]: change to isCurrentUserBlocked

File: apps/frontend/src/store/blockStore.ts [hunk 7]
```diff
@@ -22,3 +24,16 @@ export const removeUserFromBlockList = (userId: string) => {
...
+    const url = `http://[${location}]/api/v2/blocked/${userId}`;
+    return (await axios.get(calcExternalResourceLink(url))).data;
+};
+
+export const checkIfBlocked = async (contacts: GroupContact[], userId: string) => {
```
Comment (EdwardKerckhof) [inline apps/frontend/src/store/blockStore.ts:33]: this is not really checking if a contact is blocked rather filtering out blocked contacts.
I would change this to `getUnblockedContacts`

**Review by EdwardKerckhof — CHANGES_REQUESTED** [review]

File: apps/frontend/src/components/GroupManagement.vue (hunk 4 above)
Comment (JorisBassle) [reply apps/frontend/src/components/GroupManagement.vue:6]: joat

File: apps/frontend/src/components/GroupManagement.vue (hunk 5 above)
Comment (JorisBassle) [reply apps/frontend/src/components/GroupManagement.vue:19]: yes

**Review by EdwardKerckhof — APPROVED** [review]

**Pull request merged**